src/pokemon.c

```diff
@@ -0,0 +1,5567 @@
+
+        checksum = GetMonDataChecksum(&mon->box.dataBlocks, sizeof(PokemonDataBlockA) * 4);
+
+        if (checksum != mon->box.checksum) {
+            GF_ASSERT((checksum == mon->box.checksum));
+            EncryptMonData(mon->box.dataBlocks, sizeof(PokemonDataBlockA) * 4, mon->box.checksum);
+            return;
+        }
+    }
+
+    IncreaseMonDataAttribute(mon, monDataAttribute, value);
+
+    if (mon->box.partyDecrypted == 0) {
+        EncryptMonData(&mon->party, sizeof(PartyPokemon), mon->box.personality);
+        mon->box.checksum = GetMonDataChecksum(&mon->box.dataBlocks, sizeof(PokemonDataBlockA) * 4);
+        EncryptMonData(&mon->box.dataBlocks, sizeof(PokemonDataBlockA) * 4, mon->box.checksum);
+    }
+}
+
+static void IncreaseMonDataAttribute (Pokemon * mon, int monDataAttribute, int value)
+{
+    switch (monDataAttribute) {
+    case MON_DATA_CURRENT_HP:
+        if ((mon->party.unk_06 + value) > mon->party.unk_08) {
+            mon->party.unk_06 = mon->party.unk_08;
+        } else {
+            mon->party.unk_06 += value;
+        }
+        break;
+    case MON_DATA_160:
+    case MON_DATA_LEVEL:
+    case MON_DATA_162:
+    case MON_DATA_MAX_HP:
+    case MON_DATA_ATK:
+    case MON_DATA_DEF:
+    case MON_DATA_SPEED:
+    case MON_DATA_SP_ATK:
+    case MON_DATA_SP_DEF:
+    case MON_DATA_170:
+        GF_ASSERT(0);
+        break;
+    default:
+        IncreaseBoxMonDataAttribute((BoxPokemon *)&mon->box, monDataAttribute, value);
+        break;
+    }
+}
+
+static void IncreaseBoxMonDataAttribute (BoxPokemon * boxMon, int monDataAttribute, int value)
+{
+    int unused_0;
+    u16 unused_1;
+    PokemonDataBlockA * monDataBlockA;
+    PokemonDataBlockB * monDataBlockB;
+    PokemonDataBlockC * monDataBlockC;
+    PokemonDataBlockD * monDataBlockD;
+
+    monDataBlockA = (PokemonDataBlockA *)GetBoxMonDataBlock(boxMon, boxMon->personality, 0);
+    monDataBlockB = (PokemonDataBlockB *)GetBoxMonDataBlock(boxMon, boxMon->personality, 1);
+    monDataBlockC = (PokemonDataBlockC *)GetBoxMonDataBlock(boxMon, boxMon->personality, 2);
+    monDataBlockD = (PokemonDataBlockD *)GetBoxMonDataBlock(boxMon, boxMon->personality, 3);
+
+    switch (monDataAttribute) {
+    case MON_DATA_EXP:
+        if ((monDataBlockA->exp + value) > GetMonSpeciesLevelExp(monDataBlockA->species, 100)) {
+            monDataBlockA->exp = GetMonSpeciesLevelExp(monDataBlockA->species, 100);
+        } else {
+            monDataBlockA->exp += value;
+        }
+        break;
+    case MON_DATA_FRIENDSHIP:
+    {
+        int newValue;
+
+        newValue = monDataBlockA->friendship;
+
+        if ((newValue + value) > 255) {
+            newValue = 255;
+        }
+
+        if ((newValue + value) < 0) {
+            newValue = 0;
+        } else {
+            newValue += value;
+        }
+
+        monDataBlockA->friendship = newValue;
+    }
+    break;
```

Comment:
No need for a scope block here + shift `break` one indentation-level to the right.